guide/14-deep-learning/FCN_MLModel_guide-final_draft.ipynb

```diff
@@ -0,0 +1,485 @@
+{
```

Comment:
<p>Data preparation can be a time consuming process that typically involves splitting the data into training and validation sets, applying various data augmentation techniques, creating the necessary data structures for loading data into the model, memory management by using the appropriately sized mini-batches of data, and so on. </p><p>You can use <u>the</u> <code>prepare_tabulardata</code> method to create a TabularDataObject that can be fed into the <code>MLModel</code> and the <code>FullyConnectedNetwork</code> models. It allows you to use attribute<s>s</s> data from feature layers and spatial dataframes. Additionally, you can use explanatory variables from rasters. The rasters can be local rasters through the raster class or imagery layer. This method performs normalization of the datasets, imputation, and splits it into training and validation datasets before feeding <u>the datasets </u>into the model. </p><p>The <code>prepare_tabulardata</code> method <s>introspects</s> <u>inspects</u> the type of fields in the input data and automatically treats them as continuous or categorical variables. However, it also gives flexibility to overwrite this default behavior using the following example: </p><p>For example, if n predictors are used and one is categorical and the other continuous, then: </p><p><code>X = ["Field_1", ("Field_2", True),......."Field_n"]</code> </p><p>Here <strong>Field_2</strong> is treated as categorical and the others as continuous variables. </p><p>In <u>cases where </u>variables are stored as multiple rasters, then: </p><p><code>explanatory_rasters = [RasterClassObject1,(RasterClassObject2, True),.......RasterClassObjectn]</code>  </p><p>Here, RasterClassObject2 is treated as categorical and the others as continuous variables.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/718/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>